fe/fe-core/src/main/java/com/starrocks/server/MetadataMgr.java

```diff
@@ -754,34 +754,30 @@ public Statistics getTableStatistics(OptimizerContext session,
 
     public List<RemoteFileInfo> getRemoteFileInfos(Table table, List<String> partitionNames) {
         Optional<ConnectorMetadata> connectorMetadata = getOptionalMetadata(table.getCatalogName());
-        ImmutableSet.Builder<RemoteFileInfo> files = ImmutableSet.builder();
         if (connectorMetadata.isPresent()) {
             try {
-                connectorMetadata.get().getRemoteFileInfos(table, partitionNames)
-                        .forEach(files::add);
+                return connectorMetadata.get().getRemoteFileInfos(table, partitionNames);
             } catch (Exception e) {
                 LOG.error("Failed to list partition file's metadata on catalog [{}], table [{}]",
                         table.getCatalogName(), table, e);
                 throw e;
             }
         }
-        return ImmutableList.copyOf(files.build());
+        return new ArrayList<>();
```

Comment:
connectorMetadata could be empty?

Reply:
I don't understand why. maybe exception is more preferable? @stephen-shelby @Youngwb

Reply:
I think this is just defensive code.